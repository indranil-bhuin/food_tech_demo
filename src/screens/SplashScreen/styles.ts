import { StyleSheet } from 'react-native';
import { colors } from '../../../assets/colors';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        position: 'relative',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: colors.main,
        flexDirection: 'column'
    },
    imageContainer: {
        width: 300,
        height: 300,
        backgroundColor: 'white',
        borderRadius: 150
    },
    image: {
        width: 232,
        height: 206,
        left: 20,
        top: 40
    },
    splashTextContainer: {
        alignItems: 'center',
        justifyContent: 'center',
        marginTop: 80
    },
    buttonText: {
        color: '#594AB5',
        fontFamily: 'Nunito-Black',
        //fontStyle: 'normal',
        //fontWeight: '700',
        fontSize: 17,
    },
    buttonContainer: {
        //marginTop: 0
    }

})

export default styles;