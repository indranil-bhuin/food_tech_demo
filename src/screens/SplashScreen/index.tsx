import { Image, Text, TouchableOpacity, View } from 'react-native';
import React from 'react';
import styles from './styles'
import CustomText from '../../components/atoms/CustomText';
import Button from '../../components/atoms/Button';
import { colors } from '../../../assets/colors';

const SplashScreen = () => {
  return (
    <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image source={require('../../../assets/images/logo.png')} style={styles.image} />
      </View>
      <View style={styles.splashTextContainer}>
        <CustomText style={undefined} color={'#FFF'}>The Ultimate Foodcourt</CustomText>
        <CustomText style={undefined} color={'#FFF'}>for Nineleaps</CustomText>
        <CustomText style={undefined} color={'#FFF'}>for Nineleaps</CustomText>
      </View>
      <View style={styles.buttonContainer}>
        <Button
          style={undefined}
          text="Get Started"
          buttonText={styles.buttonText}
          onPress={() => console.log('Button pressed')}
          backgroundColor = {colors.white}
          />
      </View>

    </View>
  )
}

export default SplashScreen;