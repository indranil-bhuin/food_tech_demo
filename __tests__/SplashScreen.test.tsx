import React from 'react';
import renderer from 'react-test-renderer';
import SplashScreen from '../src/screens/SplashScreen';

test('SplashScreen renders correctly', () => {
  const tree = renderer.create(<SplashScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
