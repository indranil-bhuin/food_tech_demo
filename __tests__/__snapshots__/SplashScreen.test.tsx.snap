// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#594AB5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 150,
        "height": 300,
        "width": 300,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/images/logo.png",
        }
      }
      style={
        {
          "height": 206,
          "left": 20,
          "top": 40,
          "width": 232,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 80,
      }
    }
  >
    <Text
      style={
        [
          undefined,
          {
            "color": "#FFF",
            "fontFamily": "Nunito-Black",
            "fontSize": 24,
          },
        ]
      }
    >
      The Ultimate Foodcourt
    </Text>
    <Text
      style={
        [
          undefined,
          {
            "color": "#FFF",
            "fontFamily": "Nunito-Black",
            "fontSize": 24,
          },
        ]
      }
    >
      for Nineleaps
    </Text>
  </View>
  <View
    style={{}}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 25,
          "height": 70,
          "justifyContent": "center",
          "marginTop": 150,
          "opacity": 1,
          "width": 320,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;
